src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -1004,6 +1005,8 @@ InferenceProfiler::Measure(
     RETURN_IF_ERROR(manager_->GetAccumulatedClientStat(&start_stat));
   }
 
+  triton_metrics_manager_->CheckQueryingStatus();
```

Comment:
Because of how C++ exceptions and threading work, exceptions thrown in different threads do not propagate to the main thread. What this function call does here is allow the main thread to periodically check (once per pass) if the background metric querying thread has thrown an exception (e.g. wrong URL) and manually propagate it up so that user doesn't waste too much time thinking metrics are being collected when they aren't.